packages/maskbook/src/plugins/BarnBridge/UI/SmartYieldPoolView.tsx

```diff
@@ -0,0 +1,259 @@
+
+    return (
+        <Grid
+            item
+            container
+            direction="row"
+            onClick={() => onClickPoolCell(props.redirectUrl)}
+            className={classes.coinContainer}>
+            <Grid container spacing={0} direction="row" className={classes.coinPadding}>
+                <Grid container item xs={true}>
+                    <Grid item className={classes.coinText}>
+                        <Typography fontSize={13} variant="subtitle2">
+                            {props.coinName}
+                        </Typography>
+                    </Grid>
+                    <Grid item />
+                </Grid>
+                <Grid container item xs={true}>
+                    <Grid item className={classes.coinText}>
+                        <Typography fontSize={13} variant="subtitle2">
+                            {props.seniorLiquidity}
+                        </Typography>
+                    </Grid>
+                    <Grid item />
+                </Grid>
+                <Grid container item xs={true}>
+                    <Grid item className={classes.seniorAPYText}>
+                        <Typography fontSize={13} variant="subtitle2">
+                            {props.seniorAPY + '%'}
+                        </Typography>
+                    </Grid>
+                    <Grid item />
+                </Grid>
+                <Grid container item xs={true}>
+                    <Grid item className={classes.coinText}>
+                        <Typography fontSize={13} variant="subtitle2">
+                            {props.juniorLiquidity}
+                        </Typography>
+                    </Grid>
+                    <Grid item />
+                </Grid>
+                <Grid container item xs={true}>
+                    <Grid item className={classes.juniorAPYText}>
+                        <Typography fontSize={13} variant="subtitle2">
+                            {props.juniorAPY + '%'}
+                        </Typography>
+                    </Grid>
+                    <Grid item />
+                </Grid>
+            </Grid>
+        </Grid>
+    )
+}
+
+export interface SYPoolProps {
+    protocolName: string
+    coins: SYCoinProps[]
+}
+export interface SYCoinProps {
+    coinName: string
+    seniorLiquidity: string
+    seniorAPY: number
+    juniorLiquidity: string
+    juniorAPY: number
+    redirectUrl: string
+}
+
+interface SYProtocolNameProps {
+    protocolName: string
+}
+
+export function SmartYieldPoolView(props: SYPoolProps) {
+    const { classes } = useStyles()
+    const { t } = useI18N()
+    const chainId = useChainId()
+    const coins = props.coins
+
+    return (
+        <Grid container direction="column" className={classes.root}>
+            <GenerateProtocolName protocolName={props.protocolName} />
+            <GenerateHeader />
+            {props.coins.map((coin) => (
+                <GeneratePoolCell
```

Comment:
The `key` was lost here.

Reply:
I'm not sure what you mean here.... `key` in this case in `coin` and it's used to populate the GeneratePoolCell function

Reply:
You answered below -- thanks :)